src/osd/OSD.h

```diff
@@ -1692,9 +1694,8 @@ class OSD : public Dispatcher,
       }
     }; // struct ShardData
 
-    vector<ShardData*> shard_list;
     OSD *osd;
-    uint32_t num_shards;
+    ceph::tiny_vector<ShardData, 8> shards;
```

Comment:
FIXME: magic number.

@liewegas: do we have a `constexpr`-capable routine for accessing default value of a config option?

Reply:
@rzarzynski i don't think we have it. and i am afraid we won't have a `constexpr` function for accessing the default value. as `md_conifig_t::get_val<>()` involves non-constexpr expression.